agent/discover/hanacluster.go

```diff
@@ -0,0 +1,53 @@
+package discover
+
+import (
+	"fmt"
+
+	consul "github.com/hashicorp/consul/api"
+	consul_internal "github.com/trento-project/trento/internal/consul"
+)
+
+type HanaClusterDiscover struct {
+	cluster               ClusterDiscover
+	SAPid                 string
+	SAPHanaInstanceNumber string
+}
+
+// check if the current node this trento agent is running on can be discovered
+// by HanaClusterDiscover, with other words is a SAP Hana Pacemaker cluster
+func (hana HanaClusterDiscover) shouldDiscover(client consul.Client) bool {
+
+	return len(hana.SAPid) > 0
+}
+
+// Provide a Consul Key-Value Store Prefix that is used to store individual Key-Values for this Cluster
+func (hana HanaClusterDiscover) KVStorePrefix() string {
+	return fmt.Sprintf(
+		"%s/%s",
+		consul_internal.KV_CLUSTERS_PATH, hana.cluster.clusterName)
+}
+
+// Discovery module handling specifics for SAP Hana clusters
+func NewHanaClusterDiscover(client consul.Client) Discovery {
+	return func() (Discoverer, error) {
+		r := HanaClusterDiscover{}
+		//r.cluster = NewClusterDiscover(client)(client)
```

Comment:
Don't we need to uncomment this? Otherwise how is the cluster code initialized?

Reply:
well, it wasn't used. now it is, so I removed the comment and it is initialized now.